lib/client.js

```diff
@@ -92,6 +141,243 @@ function MatrixClient(opts) {
+                        }
+                    }
+                }
+            }
+        }
+        var deferred = q.defer();
+        if (devicesWithoutSession.length > 0) {
+            var queries = {};
+            for (i = 0; i < devicesWithoutSession.length; ++i) {
+                var device = devicesWithoutSession[i];
+                var query = queries[device[0]] || {};
+                queries[device[0]] = query;
+                query[device[1]] = "curve25519";
+            }
+            var path = "/keys/claim";
+            var content = {one_time_keys: queries};
+            var self = this;
+            this._http.authedRequestWithPrefix(
+                undefined, "POST", path, undefined, content,
+                httpApi.PREFIX_V2_ALPHA
+            ).then(function(res) {
+                var missing = {};
+                for (i = 0; i < devicesWithoutSession.length; ++i) {
+                    var device = devicesWithoutSession[i];
+                    var userRes = res.one_time_keys[device[0]] || {};
+                    var deviceRes = userRes[device[1]];
+                    var oneTimeKey;
+                    for (var keyId in deviceRes) {
+                        if (keyId.indexOf("curve25519:") === 0) {
+                            oneTimeKey = deviceRes[keyId];
+                        }
+                    }
+                    if (oneTimeKey) {
+                        var session = new Olm.Session();
+                        var account = new Olm.Account();
+                        try {
+                            var pickled = self.sessionStore.getEndToEndAccount();
+                            account.unpickle(self.accountKey, pickled);
+                            session.create_outbound(account, device[2], oneTimeKey);
+                            var sessionId = session.session_id();
+                            pickled = session.pickle(self.accountKey);
+                            self.sessionStore.storeEndToEndSession(
+                                device[2], sessionId, pickled
+                            );
+                        } finally {
+                            session.free();
+                            account.free();
+                        }
+                    } else {
+                        missing[device[0]] = missing[device[0]] || [];
+                        missing[device[0]].push([device[1]]);
+                    }
+                }
+                deferred.resolve({
+                    missingUsers: userWithoutDevices,
+                    missingDevices: missing
+                });
+            });
+        } else {
+            deferred.resolve({
+                missingUsers: userWithoutDevices,
+                missingDevices: []
+            });
+        }
+        this.sessionStore.storeEndToEndRoom(roomId, config);
+        return deferred.promise;
+    } else {
+        throw new Error("Unknown algorithm: " + OLM_ALGORITHM);
```

Comment:
Presumably you mean, "Unknown algorithm: "+config.algorithm. OLM_ALGORITHM being the one we do know about...